File: spring/member-server/src/main/java/com/example/memberserver/member/MemberQueryService.java
package com.example.memberserver.member;

import org.springframework.stereotype.Service;

import javax.transaction.Transactional;

@Service
public class MemberQueryService {
    private final MemberRepository memberRepository;

    public MemberQueryService(MemberRepository memberRepository) {
        this.memberRepository = memberRepository;
    }

    @Transactional
    public MemberDto find(Long memberId) {
        Member member = memberRepository.findById(memberId).orElseThrow(() -> new RuntimeException("Member not found"));
        return MemberDto.of(member);
    }
}

File: spring/member-server/src/main/kotlin/com/example/memberserver/domain/member/MemberQueryService.kt
package com.example.memberserver.domain.member

import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service

@Service
class MemberQueryService(
    private val memberRepository: MemberRepository
) {
    fun find(memberId: Long?): MemberDto {
        val member = memberRepository.findByIdOrNull(memberId) ?: throw Exception()

        return MemberDto.of(member)
    }

}
